src/Version/Version.hpp

```diff
@@ -0,0 +1,54 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+//===-------------------------- Version.hpp -------------------------------===//
+//
+// Copyright 2022 The IBM Research Authors.
+//
+// =============================================================================
+//
+// This file defines several version-related utility functions.
+//
+//===----------------------------------------------------------------------===//
+
+#pragma once
+
+#include "llvm/Support/raw_ostream.h"
+#include <string>
+
+namespace onnx_mlir {
+/// Retrieves the repository path from which onnx-mlir was built.
+std::string getOnnxMlirRepositoryPath();
+
+/// Retrieves the repository path from which LLVM was built.
+std::string getLLVMRepositoryPath();
+
+/// Retrieves the repository revision number (or identifier) from which
+/// this onnx-mlir was built.
+std::string getOnnxMlirRevision();
+
+/// Retrieves the repository revision number (or identifier) from which
+/// LLVM was built.
+std::string getLLVMRevision();
+
+/// Retrieves the full repository version that is an amalgamation of
+/// the information in getOnnxMlirRepositoryPath() and  getOnnxMlirRevision().
+/// And getLLVMRepositoryPath() and getLLVMRevision() if \p
+/// ToIncludeLLVM.
+std::string getOnnxMlirFullRepositoryVersion(bool ToIncludeLLVM);
```

Comment:
Here and elsewhere: parameters in the onnx-mlir source code start with lower case letters. Any reason to do this differently?